llpc/context/llpcCompiler.cpp

```diff
@@ -1373,17 +1453,29 @@ Result Compiler::BuildGraphicsPipeline(const GraphicsPipelineBuildInfo *pipeline
 #if LLPC_CLIENT_INTERFACE_MAJOR_VERSION < 38 || LLPC_ENABLE_SHADER_CACHE
   appCache = reinterpret_cast<IShaderCache *>(pipelineInfo->pShaderCache);
 #endif
+  Vkgc::ICache *pUserCache = nullptr;
+#if LLPC_CLIENT_INTERFACE_MAJOR_VERSION > 40
+  pUserCache = pipelineInfo->pCache;
+#endif
```

Comment:
Please remove the preprocessor directives and simplify here and in other places.